kubernetes-deployment/bootstrap.py

```diff
@@ -395,47 +414,54 @@ def remove_nodes(cluster_config, node_list_config):
 
 
 
-def option_validation(args):
+def maintain_one_node(cluster_config, maintain_config, node_config, job_name):
 
-    if args.add or args.remove:
-        if args.add and args.remove:
-            print "You could only specify one option in -a and -r"
-            return False
-        if args.file == None:
-            print "Please specify the nodelist.yaml's path"
-            return False
-        if args.deploy or args.clean:
-            print "You could not specify the option (-a or -r) with the option (-d or -c)"
-            return False
-        # Add or remove node-list
-        return True
+    module_name = "maintainlib.{0}".format(job_name)
+    module = importlib.import_module(module_name)
 
-    if args.deploy or args.clean:
+    job_class = getattr(module, job_name)
+    job_instance = job_class(cluster_config, node_config, True)
 
-        if args.deploy and args.clean:
-            print "You can only specify only one option in -d and -c !"
-            return False
+    job_instance.run()
+
+
+
+def maintain_nodes(cluster_config, node_list_config, job_name):
 
+    # Todo: load maintain from a DB such as etcd instead of a yaml file.
+    maintain_config = load_yaml_file("maintain.yaml")
+
+    for host in node_list_config['machinelist']:
+
+        maintain_one_node(cluster_config, maintain_config, node_list_config['machinelist'][host], job_name)
+
+
+
+def option_validation(args):
```

Comment:
>option_validation [](start = 4, length = 17)

please add ut. please construct typical invalid input and see if it return false and the expected error hint

Reply:
Done